userspace/falco/falco_metrics.cpp

```diff
@@ -72,9 +72,22 @@ std::string falco_metrics::to_text(const falco::app::state& state) {
 	libs::metrics::prometheus_metrics_converter prometheus_metrics_converter;
 	std::string prometheus_text;
 
-	for(auto inspector : inspectors) {
-		// Falco wrapper metrics
+	for(size_t i = 0; i < inspectors.size(); ++i) {  // Start inspector loop
+		auto& inspector = inspectors[i];
+
+		// Falco wrapper metrics, repeated for each inspector, accounting for plugins w/ event
+		// sources
 		//
+
+		/* Examples ...
```

Comment:
@sgaist this seems to be the only true reason for why we would need to loop over each inspector vs only needing the first one. The examples listed here were verified by me when testing with the dummy_c event source plugin.

Reply:
List of inspectors is created above (L64) from the state `enabled_sources`, thus we are responsible to eventually put the `syscall` inspector first in the vector :)

Reply:
Also, question: what if we have multiple source plugin enabled?
Won't
> falcosecurity_scap_engine_name_info{engine_name="source_plugin"} 1

be pushed multiple times (causing the issue that we trying to fix?)

What if we instead pushed the number of inspectors that use a certain engines, like:
```
falcosecurity_scap_engine_name_info{engine_name="source_plugin"} 4
falcosecurity_scap_engine_name_info{engine_name="bpf"} 1
```
?

Reply:
re the comment around ordering, and having the syscalls inspector at index 0 - I adjusted the approach, seems to work now.

re multiple plugin event sources -- from my perspective maybe more of a tomorrow problem and not highest priority since these less essential informational metrics compared to the more important metrics.

re not having the syscalls inspector at all, then any of the other inspectors is at index 0 and will do it's best to fetch as many metrics as it can -- all we can guarantee at this time. We did some fixes in libs a while ago, such as adding a linux lite platform so that at the very least `agent_info` and `machine_info` are available for plugin even sources only (without needing the syscalls event source).